### PR TITLE
Add two new gears.string functions: startswith, endswith

### PR DESCRIPTION
**Pull request by Veratil**
This adds the `startswith` and `endswith` functions to `gears.string`, and changes `awful.completion` to use `gears.string.startswith` instead of the local function `str_starts`.

Allows more exposure to a useful common string function.

### CODE REVIEW

File: lib/gears/string.lua [hunk 1]
```diff
@@ -106,4 +106,22 @@ function gstring.split(str, delimiter)
...
+	return string.sub(str, 1, string.len(sub)) == sub
+end
+
+--- Check if a string ends with another string
+-- @class function
+-- @name endswith
+-- @tparam string str String to search
+-- @tparam string sub String to check for
+function gstring.endswith(str, sub)
+	return sub == "" or string.sub(str,-string.len(sub)) == sub
```
Comment (psychon) [inline lib/gears/string.lua]: I'd like to see some tests for this (I'm writing this comment instead of asking why `sub == ""` is necessary, which made me notice that things are not as simple as I initially thought)

Comment (Veratil) [inline lib/gears/string.lua]: `endswith("something", "") == true`

Without the `sub == ""` check then `string.sub(str,-string.len(sub)) == sub` returns false.

Comment (Veratil) [inline lib/gears/string.lua]: That's honestly the only reason why it's there.

**Comment by actionless** [comment]
mb to use `lua-penlight` for such things? we depend on it for `luadoc` anyway

http://stevedonovan.github.io/Penlight/api/libraries/pl.stringx.html#startswith

**Comment by Veratil** [comment]
> mb to use lua-penlight for such things? we depend on it for luadoc anyway

`lua-penlight` is useful only to those who build from source, though, it's not a runtime requirement.

**Comment by actionless** [comment]
yup, you right
but also, since penlight is pure lua, it's also possible to ship static copy of it as `awful.pl` or whatever
however idk how much use of it we can make

please count my comment not as objection to merging your PR but just a topic to discuss unconditionally to this PR, it only gave me that idea

**Comment by Veratil** [comment]
Oh I'm not counting it as an objection. Just saying not everyone has lua-penlight actually installed. :)

**Comment by blueyed** [comment]
How do you feel about adding some tests for it then?

**Comment by Veratil** [comment]
> How do you feel about adding some tests for it then?

I don't see why we need to, honestly. There's literally one edge case for `endswith` which just requires the `sub == ""`.

```lua
> gstr = require('gears.string')
> gstr.startswith("something", "")
true
> gstr.startswith("something", "some")
true
> gstr.startswith("something", "none")
false
> gstr.endswith("something", "")
true
> gstr.endswith("something", "thing")
true
> gstr.endswith("something", "that")
false
```
